pubtools/pulplib/_impl/compat_attr.py

```diff
@@ -30,9 +28,7 @@ def s(*args, **kwargs):
         # don't use the attrs-generated repr by default
         kwargs["repr"] = False
 
-    if "kw_only" in kwargs and (
-        ATTR_VERSION < (18, 2) or sys.version_info < (3,)
-    ):  # pragma: no cover
+    if "kw_only" in kwargs and ATTR_VERSION < (18, 2):
         # This is only implemented for Python 3 and only in attrs 18.2 and newer.
```

Comment:
The CI fails because the statement coverage has decreased from 100%.  I'm thinking this branch is the reason why, as it was previously marked with "pragma: no cover" (which we did for py2-only paths because the coverage config uses py3) and that pragma has been removed.

How about just removing this entire branch?  I was considering this earlier in the review anyway because I'm confident there's no reason to use such an old version of attrs on python 3.  Now that it's causing this coverage issue, we may as well go ahead and remove it.

By the way, the comment at the top of this function is also now wrong and could just say something like "set some defaults when using attr.s so they cannot be easily forgotten".

Reply:
Can I also remove that function [fileds_dict](https://github.com/release-engineering/pubtools-pulplib/blob/master/pubtools/pulplib/_impl/compat_attr.py#L43) defined later in this same file then? It's also considering an earlier version of `attr` and might impact coverage as well.